rclcpp/test/rclcpp/test_any_subscription_callback.cpp

```diff
@@ -45,6 +46,20 @@ class TestAnySubscriptionCallback : public ::testing::Test
   rclcpp::MessageInfo message_info_;
 };
 
+void construct_with_null_allocator()
+{
+  // We need to wrap this in a function because `EXPECT_THROW` is a macro, and thinks
+  // that the comma in here splits macro arguments, not the template arguments.
+  rclcpp::AnySubscriptionCallback<
+    test_msgs::msg::Empty, std::allocator<void>> any_subscription_callback_(nullptr);
+}
+
+TEST(AnySubscription, null_allocator) {
+  EXPECT_THROW(
+    construct_with_null_allocator(),
+    std::runtime_error);
```

Comment:
just curious if the following works? probably bit nicer instead of introducing `construct_with_null_allocator`?

```suggestion
    {
      rclcpp::AnySubscriptionCallback<
        test_msgs::msg::Empty, std::allocator<void>> any_subscription_callback_(nullptr);
    }, std::runtime_error);
```

Reply:
It ends up having the same problem.  I think the macro parser is dumb and just assumes that the next comma in there is splitting arguments, regardless what is there in front.

I could probably do something with token pasting, but in the end I thought that having a separate function was cleaner-looking.

But thanks anyway for the idea and the review.  I'm going to go ahead and run CI on this.